playwright-e2e/dsm/component/navigation/navigation.ts

```diff
@@ -64,6 +65,52 @@ export class Navigation {
     await this.selectFrom(MainMenuEnum.MISCELLANEOUS, miscName);
   }
 
+  public async getDisplayedMainMenu(): Promise<MainMenuEnum[]> {
+    let menuOptions: Locator[] = [];
+    await expect(async () => {
+      menuOptions = await this.page.locator(`//app-navigation//ul[1]//li[contains(@class, 'dropdown')]/a`).all();
+      const countOfMenuOptions = menuOptions.length;
+      logInfo(`Amount of main menu options: ${countOfMenuOptions}`);
+      expect(countOfMenuOptions).toBeGreaterThanOrEqual(1);
+    }).toPass({
+      intervals: [5_000]
+    });
+
+    const displayedMenuOptions: MainMenuEnum[] = [];
+    const selectedStudyText = MainMenuEnum.SELECTED_STUDY as string;
+    for (const option of menuOptions) {
+      let name = (await option.innerText()).trim();
+      if (name.startsWith(selectedStudyText)) {
+        name = selectedStudyText;
+      }
+      logInfo(`Main menu option: ${name}`);
+      displayedMenuOptions.push(name as MainMenuEnum);
+    }
+    return displayedMenuOptions;
+  }
+
+  public async getDisplayedSamplesMenuOptions(): Promise<SamplesNavEnum[]> {
```

Comment:
Here is a generic func in `dropdown.ts`: (replace existing `getAllOptions` function with this new generic func)

```
   async getDisplayedOptions<T extends MiscellaneousEnum | SamplesNavEnum | StudyNavEnum | StudyEnum>(): Promise<T[]> {
    await this.open();
    const options: T[] = [];
    const links: Locator[] = await this.toLocator().locator('ul.dropdown-menu').locator('a').all();
    for (const link of links) {
      const innerText = (await link.innerText()).trim();
      options.push(innerText as T);
    }
    return options;
  }

```

Usage example:

```
const displayedMenuOptions = await samplesMenu.getDisplayedOptions<SamplesNavEnum>();
console.log(displayedMenuOptions)
```